src/commands/flutter-symbols/upload.ts

```diff
@@ -0,0 +1,468 @@
+    apiKeyValidator: ApiKeyValidator
+  ): Promise<UploadStatus[]> {
+    const files = this.getFlutterSymbolFiles(this.dartSymbolsLocation!)
+
+    const filesMetadata = files.map((filename) => ({filename, ...getArchInfoFromFilename(filename)}))
+
+    const requestBuilder = getFlutterRequestBuilder(this.config.apiKey!, this.cliVersion, this.config.datadogSite)
+    try {
+      const results = await asyncPool(this.maxConcurrency, filesMetadata, async (fileMetadata) => {
+        if (!fileMetadata.arch || !fileMetadata.platform) {
+          renderFailedUpload(
+            fileMetadata.filename,
+            'Skipped because we could not determine the architecture or platform.'
+          )
+
+          return UploadStatus.Skipped
+        }
+
+        if (this.dryRun) {
+          this.context.stdout.write(`[DRYRUN] ${renderUpload('Dart Symbol File', fileMetadata.filename!)}`)
+
+          return UploadStatus.Success
+        }
+
+        const metadata = this.getFlutterMetadata(fileMetadata.platform, fileMetadata.arch)
+        const payload = {
+          content: new Map<string, MultipartValue>([
+            ['event', {value: JSON.stringify(metadata), options: {filename: 'event', contentType: 'application/json'}}],
+            [
+              VALUE_NAME_DART_MAPPING,
+              {value: fs.createReadStream(fileMetadata.filename), options: {filename: DART_SYMBOL_FILE_NAME}},
+            ],
+          ]),
+        }
+        if (this.gitData !== undefined) {
+          payload.content.set('repository', this.getGitDataPayload()!)
+        }
+
+        return uploadMultipartHelper(requestBuilder, payload, {
+          apiKeyValidator,
+          onError: (e) => {
+            this.context.stdout.write(renderFailedUpload(fileMetadata.filename!, e.message))
+            metricsLogger.logger.increment('failed', 1)
+          },
+          onRetry: (e, attempts) => {
+            this.context.stdout.write(renderRetriedUpload(fileMetadata.filename, e.message, attempts))
+            metricsLogger.logger.increment('retries', 1)
+          },
+          onUpload: () => {
+            this.context.stdout.write(renderUpload('Flutter Symbol File', fileMetadata.filename))
+          },
+          retries: 5,
+        })
+      })
+
+      return results
+    } catch (error) {
+      throw error
+    } finally {
+      try {
+        await metricsLogger.flush()
+      } catch (err) {
+        this.context.stdout.write(`WARN: ${err}\n`)
+      }
+    }
+  }
+
+  private async performDsymUpload() {
+    const dsymUploadCommand = ['dsyms', 'upload', this.iosDsymsLocation!]
+    if (this.dryRun) {
+      dsymUploadCommand.push('--dry-run')
+    }
+
+    const exitCode = await performSubCommand(dsyms.UploadCommand, dsymUploadCommand, this.context)
+    if (exitCode && exitCode !== 0) {
+      return UploadStatus.Failure
+    }
+
+    return UploadStatus.Success
+  }
+
+  private async verifyParameters(): Promise<boolean> {
+    if (!this.serviceName) {
```

Comment:
To improve the UX it would be nice to display all errors at once (instead of requiring the user to re-run the command after having fixed one issue at a time).

Reply:
Yeah, that's a good call, and should be fairly easy to implement.